include/sys/arch_interface.h

```diff
@@ -328,6 +328,29 @@ int arch_irq_connect_dynamic(unsigned int irq, unsigned int priority,
  * @see ISR_DIRECT_DECLARE()
  */
 
+/**
+ * Arch-specific hook for allocating IRQs
+ *
+ * @return The newly allocated IRQ or UINT_MAX on error.
+ */
+unsigned int arch_irq_allocate(void);
+
+/**
+ * Arch-specific hook for declaring an IRQ being used
+ *
+ * @param irq the IRQ to declare being used
+ */
+void arch_irq_on_use(unsigned int irq);
```

Comment:
"arch_irq_set_used(bool)" maybe?

Reply:
you mean arch_irq_set_used(irq), yes its' a better name.